python/dgl/graphbolt/minibatch.py

```diff
@@ -290,27 +275,106 @@ def to_dgl_blocks(self):
                             dgl.EID
                         ] = reverse_ids
         else:
-            # Assign node features to the outermost layer's source nodes.
-            if self.node_features:
-                for feature_name, feature in self.node_features.items():
-                    blocks[0].srcdata[feature_name] = feature
-            # Assign edge features.
-            if self.edge_features:
-                for block, edge_feature in zip(blocks, self.edge_features):
-                    for feature_name, feature in edge_feature.items():
-                        block.edata[feature_name] = feature
             blocks[0].srcdata[dgl.NID] = self.sampled_subgraphs[
                 0
             ].original_row_node_ids
             # Assign reverse edges ids.
             for block, subgraph in zip(blocks, self.sampled_subgraphs):
                 if subgraph.original_edge_ids is not None:
                     block.edata[dgl.EID] = subgraph.original_edge_ids
-
         return blocks
 
-    def __repr__(self) -> str:
-        return _minibatch_str(self)
+    def to_dgl(self):
+        """ "Converting a `MiniBatch` into a DGL MiniBatch that contains
+        everything necessary for computation."
+        """
+        minibatch = DGLMiniBatch(
+            blocks=self._to_dgl_blocks(),
+            input_nodes=self.input_nodes,
+            output_nodes=self.seed_nodes,
+            node_features=self.node_features,
+            edge_features=self.edge_features,
+            labels=self.labels,
+        )
+        assert (
+            minibatch.blocks is not None
+        ), "Sampled subgraphs for computation are Missing."
+
+        # For link prediction tasks.
+        if self.compacted_node_pairs is not None:
+            assert (
+                self.compacted_node_pairs is not None
+            ), "Evaluate graph for Computation in link prediction task is Missing."
```

Comment:
if and then assert? Does this assert actually required?

Reply:
No，should be removed